tests/unit/test_characterProcessing.py

```diff
@@ -0,0 +1,39 @@
+# A part of NonVisual Desktop Access (NVDA)
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+# Copyright (C) 2020 NV Access Limited
+
+"""Unit tests for the characterProcessing module.
+"""
+
+import unittest
+import re
+from characterProcessing import SYMLVL_ALL
+from characterProcessing import SpeechSymbolProcessor
+from characterProcessing import processSpeechSymbols as process
+
+
+class TestComplex(unittest.TestCase):
+	"""Test the complex symbols rules.
+	"""
+
+	def _replace(self, m):
+		group = m.lastgroup
+		if group == 'date':

```

Comment:
Why is this specific to date?

Reply:
It is just the name of the group below, I could have put `bar` there and put `(?P<bar>` instead of `(?P<date>`, or whatever foobar-like name you'd prefer.